dataSource1/src/main/java/QuizGateway.kt

```diff
@@ -0,0 +1,4 @@
+interface QuizGateway {
+    fun getSessionToken(): String
+    fun getQuestions(amount: Int = 10, category: String, difficulty: String, type: String): String
+}
```

Comment:
if you place this files to `datasource` package please name it ...Datasource instead of Gateway

Reply:
+